docs/synapse/userguides/ug015_storm_ref_pivot.rst

```diff
@@ -11,85 +15,339 @@ Where specific query examples are given, they are commonly provided in pairs usi
+  ::
+    ask #cno.threat.t5.tc totags()
+
+* ``totags()`` and ``fromtags()`` are often used together to:
+  
+  * pivot from a set of nodes, to the tags applied to those nodes, to other nodes that have the same tags; or
+  * from a set of tags, to nodes those tags are applied to, to other tags applied to those same nodes.
 
 jointags()
 ----------
-Todo
+Returns all specified nodes that have **any** of the tags applied to **any** of the working set of nodes.
+
+``jointags()`` can be thought of as executing a ``totags()`` operation followed by a ``fromtags()`` operation.
+
+Optional parameters:
+
+* **<form>:** return only nodes of the specified form(s).
+* If no forms are specified, ``jointags()`` returns all nodes for all forms to which the tags are applied.
+
+**Operator syntax:**
+
+.. parsed-literal::
+  
+  **jointags(** [ *<form_1>* **,** *<form_2>* **,** *...<form_n>* ] **)**
+
+**Macro syntax:**
+
+N/A
+
+**Examples:**
+
+* Return the set of nodes that share any of the tags applied to the working set of nodes:
+  ::
+    jointags()
+
+* Return the set of ``inet:fqdn`` and ``inet:email`` nodes that share any of the tags applied to the working set of nodes:
+  ::
+    jointags( inet:fqdn, inet:email )
+
+**Usage notes:**
+
+* ``jointags()`` pivots using the set of leaf tags only. For example if nodes in the working set have the tag ``#foo.bar.baz``, ``jointags()`` will return other nodes with ``#foo.bar.baz``, but not nodes with ``#foo.bar`` or ``#foo`` alone.
+
+* ``jointags()``, like ``refs()``, can be useful to "explore" other nodes that share some analytical assessment (tag) with the working set of nodes, but may return a large number of nodes. It may be more efficient to narrow the scope of the query using ``totags()`` in combination with a filter operator (e.g., to potentially limit the specific tags selected) followed by ``fromtags()``.
 
 tree()
 ------
 
-The `tree()` operator acts as a recursive pivot for forms which reference their own types or multiple duplicate ptypes.
+Recursively return the set of nodes that have a property type (*<type> = <valu>* or *<ptype> = <valu>*) that matches the specified property type (*<type> = <valu>* or *<ptype> = <valu>*) from the working set of nodes.
 
-**Operator Syntax:**
+``tree()`` can be thought of as a recursive pivot that can be used to "traverse" a set of nodes which reference their own types or multiple duplicate ptypes (such as domains / subdomains, or tags in a tag hierarchy). This allows a user to build a set of nodes which have self-referencing forms. The recursive pivot takes the place of multiple single pivots.
 
-.. parsed-literal::
+Optional parameters:
```

Comment:
`limit` also works on tree (limit number of nodes vs recurnlim which is a limit on the number of recursive queries to run)

Reply:
Fixed. Also added note that recurnlim is currently ignored due to a known bug.